edx_django_utils/monitoring/internal/code_owner/middleware.py

```diff
@@ -43,105 +41,85 @@ def process_exception(self, request, exception):
 
     def _set_code_owner_attribute(self, request):
         """
-        Sets the code_owner custom attribute, as well as several supporting custom attributes.
-
-        See CodeOwnerMonitoringMiddleware docstring for a complete list of attributes.
-
+        Sets the code_owner custom attribute for the request.
         """
-        code_owner, path_error = self._set_code_owner_attribute_from_path(request)
-        if code_owner:
-            set_custom_attribute('code_owner', code_owner)
-            return
-        if not path_error:
-            # module found, but mapping wasn't configured
-            code_owner = self._set_code_owner_attribute_catch_all()
+        module = self._get_module_from_request(request)
+        if module:
+            code_owner = get_code_owner_from_module(module)
             if code_owner:
                 set_custom_attribute('code_owner', code_owner)
-            return
+                return
 
-        code_owner, transaction_error = self._set_code_owner_attribute_from_current_transaction(request)
+        code_owner = _get_catch_all_code_owner()
         if code_owner:
             set_custom_attribute('code_owner', code_owner)
-            return
-        if not transaction_error:
-            # transaction name found, but mapping wasn't configured
-            code_owner = self._set_code_owner_attribute_catch_all()
-            if code_owner:
-                set_custom_attribute('code_owner', code_owner)
-            return
 
-        code_owner = self._set_code_owner_attribute_catch_all()
-        if code_owner:
-            set_custom_attribute('code_owner', code_owner)
-            return
+    def _get_module_from_request(self, request):
+        """
+        Get the module from the request path or the current transaction.
+
+        Side-effects:
+            Sets code_owner_module custom attribute, used to determine code_owner.
+            If module was not found, may set code_owner_path_error and/or
+                code_owner_transaction_error custom attributes if applicable.
+
+        Returns:
+            str: module name or None if not found
+
+        """
+        if not is_code_owner_mappings_configured():
+            return None
+
+        module, path_error = self._get_module_from_request_path(request)
+        if module:
+            set_custom_attribute('code_owner_module', module)
+            return module
+
+        module, transaction_error = self._get_module_from_current_transaction()
+        if module:
+            set_custom_attribute('code_owner_module', module)
+            return module
 
-        # only report errors if code_owner couldn't be found, including catch-all
+        # monitor errors if module was not found
         if path_error:
             set_custom_attribute('code_owner_path_error', path_error)
         if transaction_error:
             set_custom_attribute('code_owner_transaction_error', transaction_error)
```

Comment:
**Reviewer Note**: This refactor ended up with a slight change in logic, where these error attributes are created even when we have a catch-all defined for code_owner. It's probably better this way if you actually want to find issues that resulted in the catch-all. :)